### PR TITLE
Ignore non-running pods in inspectors

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: inspectors/altnamesforpod/altnamesforpod.go [hunk 1]
```diff
@@ -45,12 +45,12 @@ func (a *altnamesforpod) Inspect(client kubernetes.Interface, request *certifica
 		return msg, nil
 	}
 
-	podList, err := client.CoreV1().Pods(namespace).List(metaV1.ListOptions{FieldSelector: "status.podIP=" + podIp})
+	podList, err := client.CoreV1().Pods(namespace).List(metaV1.ListOptions{FieldSelector: "status.phase=running,status.podIP=" + podIp})
 	if err != nil {
 		return "", err
 	}
 	if len(podList.Items) == 0 {
```
Comment (rajatjindal) [inline inspectors/altnamesforpod/altnamesforpod.go]: can you print debug msg when list > 1

File: inspectors/subjectispodforuser/subjectispodforuser.go [hunk 2]
```diff
@@ -37,17 +37,17 @@ func (s *subjectispodforuser) Inspect(client kubernetes.Interface, request *cert
 		return msg, nil
 	}
 
-	podList, err := client.CoreV1().Pods(namespace).List(metaV1.ListOptions{FieldSelector: "status.podIP=" + podIp})
+	podList, err := client.CoreV1().Pods(namespace).List(metaV1.ListOptions{FieldSelector: "status.phase=running,status.podIP=" + podIp})
 	if err != nil {
 		return "", err
 	}
 	if len(podList.Items) == 0 {
```
Comment (rajatjindal) [inline inspectors/subjectispodforuser/subjectispodforuser.go]: same here, can you please add debug msg if list > 1

File: inspectors/altnamesforpod/altnamesforpod.go [hunk 3]
```diff
@@ -45,12 +46,15 @@ func (a *altnamesforpod) Inspect(client kubernetes.Interface, request *certifica
...
 	}
 
-	podList, err := client.CoreV1().Pods(namespace).List(metaV1.ListOptions{FieldSelector: "status.podIP=" + podIp})
+	podList, err := client.CoreV1().Pods(namespace).List(metaV1.ListOptions{FieldSelector: "status.phase in (pending,running),status.podIP=" + podIp})
 	if err != nil {
 		return "", err
 	}
 	if len(podList.Items) == 0 {
-		return fmt.Sprintf("No POD in namespace %q with IP %q", namespace, podIp), nil
+		return fmt.Sprintf("No running POD in namespace %q with IP %q", namespace, podIp), nil
```
Comment (rajatjindal) [inline inspectors/altnamesforpod/altnamesforpod.go]: change this to pending/running

File: inspectors/subjectispodforuser/subjectispodforuser.go [hunk 4]
```diff
@@ -37,17 +38,20 @@ func (s *subjectispodforuser) Inspect(client kubernetes.Interface, request *cert
...
 	}
 
-	podList, err := client.CoreV1().Pods(namespace).List(metaV1.ListOptions{FieldSelector: "status.podIP=" + podIp})
+	podList, err := client.CoreV1().Pods(namespace).List(metaV1.ListOptions{FieldSelector: "status.phase in (pending,running),status.podIP=" + podIp})
 	if err != nil {
 		return "", err
 	}
 	if len(podList.Items) == 0 {
-		return fmt.Sprintf("No POD in namespace %q with IP %q", namespace, podIp), nil
+		return fmt.Sprintf("No running POD in namespace %q with IP %q", namespace, podIp), nil
```
Comment (rajatjindal) [inline inspectors/subjectispodforuser/subjectispodforuser.go]: pending/running POD

**Review by rajatjindal — APPROVED** [review]

**Review by rajatjindal — APPROVED** [review]

**Review by raghu67 — APPROVED** [review]
LGTM

**Pull request merged**